target/scripts/startup/setup.d/security/rspamd.sh

```diff
@@ -225,3 +223,76 @@ function __rspamd__handle_modules_configuration
     done < <(_get_valid_lines_from_file "${RSPAMD_CUSTOM_COMMANDS_FILE}")
   fi
 }
+
+# This function adds a Sieve script to automatically move e-mails that Rspamd marked
+# with the `X-Spam: Yes` header to the "Junk" folder.
+function __rspamd__setup_move_to_junk_automatically
+{
+  __rspamd__log 'debug' 'Spam will automatically be moved to the "Junk" folder'
+
+  # In `/etc/dovecot/conf.d/90-sieve.conf`, sieve_after is set to
+  # `/usr/lib/dovecot/sieve-global/after/`.
+  local DOVECOT_SIEVE_AFTER_DIR='/usr/lib/dovecot/sieve-global/after'
+  local SPAM_TO_FOLDER_SIEVE_FILE="${DOVECOT_SIEVE_AFTER_DIR}/spam-to-folder.sieve"
+
+  mkdir -p "${DOVECOT_SIEVE_AFTER_DIR}"
+  cat >"${SPAM_TO_FOLDER_SIEVE_FILE}" << EOF
+require ["fileinto","mailbox"];
+
+if header :contains "X-Spam" "Yes" {
+    fileinto :create "Junk";
+}
+
+EOF
+
+  sievec "${SPAM_TO_FOLDER_SIEVE_FILE}"
+}
+
+# This function sets up intelligent learning of Junk, by
+#
+# 1. enabling auto-learn for the classifier-bayes module
+# 2. setting up sieve scripts that detect when a user is moving e-mail
+#    from or to the "Junk" folder, and learning them as ham or spam.
+function __rspamd__setup_learning
+{
+  __rspamd__log 'debug' 'Setting up intelligent learning of spam and ham'
+
+  local SIEVE_CONFIGURATION_FILE='/etc/dovecot/conf.d/90-sieve.conf'
+  local SIEVE_PIPE_BIN_DIR='/usr/lib/dovecot/sieve-pipe'
+
+  sed -i -E 's|(mail_plugins =.*)|\1 imap_sieve|' /etc/dovecot/conf.d/20-imap.conf
```

Comment:
I guess a file change here is expected. If so, I advice to consequently use `sedfile -i` instead of `sed -i`